common/djangoapps/student/models.py

```diff
@@ -1865,13 +1861,12 @@ def refundable(self, user_already_has_certs_for=None):
         if getattr(self, 'can_refund', None) is not None:
             return True
 
-        # If the student has already been given a certificate they should not be refunded
-        if user_already_has_certs_for is not None:
-            if self.course_id in user_already_has_certs_for:
-                return False
-        else:
-            if GeneratedCertificate.certificate_for_student(self.user, self.course_id) is not None:
-                return False
+        # If the student has already been given a certificate in a non refundable status they should not be refunded
+        certificate = GeneratedCertificate.certificate_for_student(self.user, self.course_id)
```

Comment:
I had already looked at removing the direct usage of the `GeneratedCertificate` model here during my initial investigation and ran into a circular import issue. I didn't look into solving it too deeply. Is this something we should fix now? Maybe we should open a ticket to revisit after refunding is fixed?